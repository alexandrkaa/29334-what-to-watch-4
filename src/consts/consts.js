export const MovieCardFullTabsIds = {
  OVERVIEW: `OVERVIEW`,
  DETAILS: `DETAILS`,
  REVIEWS: `REVIEWS`,
};
export const PreviewVideoSize = {
  width: 280,
  height: 175
};

export const MovieCardFullTabs = [
  {
    id: `OVERVIEW`,
    name: `Overview`,
  },
  {
    id: `DETAILS`,
    name: `Details`
  },
  {
    id: `REVIEWS`,
    name: `Reviews`
  },
];

export const SECS_PER_HOUR = 3600;
export const MOVIES_LIKE_THIS_NUM = 4;
export const DEFAULT_GENRE = `All genres`;
export const MOVIES_LIMIT = 8;
export const GENRES_LIMIT = 8;
export const START_PREVIEW_DELAY = 1000;
export const BASE_URL = `https://htmlacademy-react-3.appspot.com`;
export const BASE_API_URL = `${BASE_URL}/wtw`;
export const API_REQUEST_TIMEOUT = 5000;

export const FiledsIds = {
  EMAIL_FIELD_ID: `user-email`,
  PASSWORD_FIELD_ID: `user-password`,
  RATING_FIELD_ID: `rating`,
  COMMENTS_FIELD_ID: `comments`,
};

export const SignInFields = [
  {
    id: FiledsIds.EMAIL_FIELD_ID,
    label: `Email address`,
    type: `email`,
    placeholder: `Email address`,
  },
  {
    id: FiledsIds.PASSWORD_FIELD_ID,
    label: `Password`,
    type: `password`,
    placeholder: `Password`,
  },
];

export const NetworkErrors = {
  UNAUTHORIZED: 401,
  BAD_REQUEST: 400,
};

export const AppRoutes = {
  MAIN_PAGE: `/`,
  FULL_PLAYER_PAGE: `/player`,
  FILM_PAGE: `/films`,
  LOGIN_PAGE: `/login`,
  REVIEW_PAGE: `/review`,
};

export const REVIEW_STARS_NUMBER = 5;

export const ReviewTextLength = {
  MIN: 1,
  MAX: 400,
};

export const ComponentsKeys = {
  BREADCRUMBS: `Breadcrumbs`,
  USERPROFILE: `UserProfile`,
};
