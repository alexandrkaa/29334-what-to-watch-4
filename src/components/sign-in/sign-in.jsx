import React from 'react';
import Header from '../header/header.jsx';
import Footer from '../footer/footer.jsx';
import InputField from '../input-filed/input-field.jsx';
import {SignInFields} from '../../consts/consts.js';
import {isValidField} from '../../utils/common.js';

class SignIn extends React.PureComponent {
  constructor(props) {
    super(props);
    const initialState = {};
    SignInFields.forEach((fld) => {
      initialState[fld.id] = {
        value: ``,
        isValid: false
      };
    });
    this.state = initialState;
    this._handleInputChange = this._handleInputChange.bind(this);
  }

  _handleInputChange(id, value) {
    this.setState({
      [id]: {
        value,
        isValid: isValidField(id, value)
      }
    });
  }

  _handleFormSubmit(evt) {
    evt.preventDefault();
    // const _keys = Object.keys(this.state);
    let isFormValid = true;
    for (const key in this.state) {
      if (this.state[key].isValid === false) {
        isFormValid = false;
        break;
      }
    }
  }

  render() {
    return (
      <div className="user-page">

        <Header headerClassName="user-page__head" >
          {<h1 className="page-title user-page__title">Sign in</h1>}
        </Header>

        <div className="sign-in user-page__content">
          {/* <div className="sign-in__message">
            <p>We canâ€™t recognize this email <br /> and password combination. Please try again.</p>
          </div> */}
          <form action="#" onSubmit={_handleFormSubmit} className="sign-in__form">
            <div className="sign-in__fields">
              {
                SignInFields.map((fld) => {
                  return (
                    <InputField
                      key={fld.id}
                      id={fld.id}
                      label={fld.label}
                      type={fld.type}
                      placeholder={fld.placeholder}
                      value={this.state[fld.id].value}
                      onChange={this._handleInputChange}
                    />
                  );
                })
              }
            </div>
            <div className="sign-in__submit">
              <button className="sign-in__btn" type="submit">Sign in</button>
            </div>
          </form>
        </div>

        <Footer />

      </div>
    );
  }
}

export default SignIn;
