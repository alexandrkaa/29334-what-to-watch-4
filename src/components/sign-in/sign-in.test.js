import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import {SignInFields} from '../../consts/consts.js';
import {SignIn} from './sign-in.jsx';
const renderer = new ShallowRenderer();

it(`<SignIn /> should match snapshot`, () => {
  const userEmail = {
    value: `qq@qq.ru`,
    isValid: true,
  };

  const userPassword = {
    value: `123456`,
    isValid: true,
  };

  const onInputChange = jest.fn();
  const onFormSubmit = jest.fn();

  const loginStatusCode = 200;
  renderer.render(
      <SignIn
        loginStatusCode={loginStatusCode}
        signInFields={SignInFields}
        user-email={userEmail}
        user-password={userPassword}
        isLoading={false}
        isAuthorized={false}
        onInputChange={onInputChange}
        onFormSubmit={onFormSubmit}
      />
  );
  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
