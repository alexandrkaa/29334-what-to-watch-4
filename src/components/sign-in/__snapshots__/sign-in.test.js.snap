// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIn /> should match snapshot <SignIn /> should match snapshot when auth 1`] = `
<Redirect
  to="/"
/>
`;

exports[`<SignIn /> should match snapshot <SignIn /> should match snapshot when no auth 1`] = `
<div
  className="user-page"
>
  <Memo(Header)
    headerClassName="user-page__head"
  >
    <h1
      className="page-title user-page__title"
    >
      Sign in
    </h1>
    <Memo(Connect(UserProfile)) />
  </Memo(Header)>
  <div
    className="sign-in user-page__content"
  >
    <form
      className="sign-in__form"
      onSubmit={[Function]}
    >
      <div
        className="sign-in__fields"
      >
        <InputField
          id="user-email"
          isDisabled={false}
          label="Email address"
          onChange={[MockFunction]}
          placeholder="Email address"
          type="email"
          value="qq@qq.ru"
        />
        <InputField
          id="user-password"
          isDisabled={false}
          label="Password"
          onChange={[MockFunction]}
          placeholder="Password"
          type="password"
          value="123456"
        />
      </div>
      <div
        className="sign-in__submit"
      >
        <button
          className="sign-in__btn"
          disabled={false}
          type="submit"
        >
          Sign in
        </button>
      </div>
    </form>
  </div>
  <Memo(Footer) />
</div>
`;
